corehq/apps/es/tests/test_client.py

```diff
@@ -346,14 +346,27 @@ def test_indices_refresh(self):
         doc_adapter = TestDocumentAdapter()
         doc_adapter.index = self.index  # use our index so it gets cleaned up
         self.adapter.index_create(self.index)
-        doc_adapter.upsert(TestDoc("1", "test"))
+        self.adapter = ElasticManageAdapter()
+
+        def set_refresh_interval(value):
+            self.adapter._index_put_settings(
+                self.index,
+                {"index.refresh_interval": value}
+            )
 
         def get_search_hits():
             return doc_adapter.search({})["hits"]["hits"]
 
-        self.assertEqual([], get_search_hits())
-        self.adapter.indices_refresh([doc_adapter.index])
-        docs = [h["_source"] for h in get_search_hits()]
+        # Disable auto-refresh to ensure the index doesn't refresh between our
+        # upsert and search (which would cause this test to fail).
+        set_refresh_interval("-1")
+        try:
+            doc_adapter.upsert(TestDoc("1", "test"))
+            self.assertEqual([], get_search_hits())
+            self.adapter.indices_refresh([doc_adapter.index])
+            docs = [h["_source"] for h in get_search_hits()]
+        finally:
+            set_refresh_interval("5s")  # set it back to previous value
```

Comment:
nit: How does this test know what the previous value was? And what if it changes?

If this is a pattern that may be useful in other tests it might make sense to extract it to a context manager utility function, which would be used like
```py
with suspend_view_refresh(self.adapter):
    doc_adapter.upsert(TestDoc("1", "test"))
    self.assertEqual([], get_search_hits())
    ...
```

Reply:
It doesn't.  In fact the test doesn't need to set anything back because the index is destroyed immediately after (this is an index-specific setting).  I'm going to remove this.